internal/federationdomain/endpoints/token/token_handler_test.go

```diff
@@ -1699,12 +1771,21 @@ func TestTokenEndpointTokenExchange(t *testing.T) { // tests for grant_type "urn
 			// Also assert which are the different from the original downstream ID token.
 			requireClaimsAreNotEqual(t, "jti", claimsOfFirstIDToken, tokenClaims) // JWT ID
 			requireClaimsAreNotEqual(t, "aud", claimsOfFirstIDToken, tokenClaims) // audience
-			requireClaimsAreNotEqual(t, "exp", claimsOfFirstIDToken, tokenClaims) // expires at
-			require.Greater(t, tokenClaims["exp"], claimsOfFirstIDToken["exp"])
 			requireClaimsAreNotEqual(t, "iat", claimsOfFirstIDToken, tokenClaims) // issued at
 			require.Greater(t, tokenClaims["iat"], claimsOfFirstIDToken["iat"])
+			requireClaimsAreNotEqual(t, "exp", claimsOfFirstIDToken, tokenClaims) // expires at
+			if test.authcodeExchange.want.wantIDTokenLifetimeSeconds == 0 {
+				// If the ID token lifetime of the original ID token was not customized by configuration,
```

Comment:
Is there a two minute delay in the test? I should go look. I would imagine that this assertion is always true: for a specific client, a new token will always `exp` after an earlier token. I think understanding this test probably means I need to look at this whole test. If the new token is a refreshed token, could we call that `refreshedToken`? Otherwise it's not clear from this diff whether the new token is a refreshed token with exactly two minute lifetime or a result of a new authorization with the configured lifetime.

Reply:
There is a one second sleep in this test to make sure that the new ID token returned by the token exchange will be more obviously different than the original ID token in its timestamp-related claims.

The lifetime of the original ID token could be the default two minutes, or it could be configured to be 4242 seconds, as is done in one test of the test table. The lifetime of the new ID token returned by the token exchange will always be 2 minutes (not configurable).

In the 4242 case, I do an authcode flow and my original ID token expires 4242 seconds (~70 minutes) from now. Then I wait one second. Then I do the token exchange and get a new ID token which expires 2 minutes from now. So in this case, my original ID token is valid for much longer than my new one.

The assertion was previously assuming that both tokens will have a 2 minute lifetime, with ~1 second delay in between when they were issued, in which case the newer token should always expire after the older token.